### PR TITLE
feat: prosody plugin for sending system chat messages

### PR DESCRIPTION
**Pull request by quitrk**

### CODE REVIEW

File: react/features/chat/constants.ts
Comment (saghul) [inline react/features/chat/constants.ts:50]: MESSAGE_TYPE_SYSTEM, for consistency?

File: resources/prosody-plugins/mod_muc_system_chat_message.lua
Comment (saghul) [inline resources/prosody-plugins/mod_muc_system_chat_message.lua:54]: Check if payload is nil and return 400

**Review by saghul — COMMENTED** [review]
Good stuff! Left some comments, PTAL. @damencho can you PTAL at the Prosody code too please?

**Review by saghul — DISMISSED** [review]
LGTM! Pl wait for @damencho 's feedback before merging.

File: resources/prosody-plugins/mod_muc_system_chat_message.lua
Comment (damencho) [inline resources/prosody-plugins/mod_muc_system_chat_message.lua:17]: Please use cjson.

File: resources/prosody-plugins/mod_muc_system_chat_message.lua
Comment (damencho) [inline resources/prosody-plugins/mod_muc_system_chat_message.lua:7]: Is there a reason this to be global, as you are listening only for the muc component. You can be loading just there.

File: resources/prosody-plugins/mod_muc_system_chat_message.lua
Comment (damencho) [inline resources/prosody-plugins/mod_muc_system_chat_message.lua:45]: Will this be used a lot? Will this be chatty? 
Can you make it debug, or at least move it and print in which room was that used, otherwise this log means nothing when debugging what happens in a room.

**Review by damencho — CHANGES_REQUESTED** [review]

File: resources/prosody-plugins/mod_muc_system_chat_message.lua
Comment (quitrk) [reply resources/prosody-plugins/mod_muc_system_chat_message.lua:45]: ill mark it as debug, since this was the original purpose

Comment (damencho) [reply resources/prosody-plugins/mod_muc_system_chat_message.lua:45]: Look at the other modules and how debug prints are behind a flag, that is for performance purposes. 
As strings and replacements are done even when not printed, although its not on a hot path its good to have code consistent :) Thanks.

**Review by damencho — APPROVED** [review]

**Review by saghul — APPROVED** [review]

**Pull request merged**
